### PR TITLE
flibcpp: add new version and smoke test

### PR DESCRIPTION
**Pull request by sethrj**

### CODE REVIEW

File: var/spack/repos/builtin/packages/flibcpp/package.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ class Flibcpp(CMakePackage):
     git = "https://github.com/swig-fortran/flibcpp.git"
     url = "https://github.com/swig-fortran/flibcpp/archive/v0.3.1.tar.gz"
```
Comment (aprokop) [inline var/spack/repos/builtin/packages/flibcpp/package.py]: I know it's not necessary, but wanna update url to 1.0.1 too?

**Review by aprokop — COMMENTED** [review]
Looks fine to me, but I have not tested it.

**Review by alalazo — CHANGES_REQUESTED** [review]
Style tests are failing

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Fyi @tldahlgren

**Pull request merged**